nevergrad/functions/powersystems/core.py

```diff
@@ -6,52 +6,50 @@
 # This code is based on a code and ideas by Emmanuel Centeno and Antoine Moreau,
 # University Clermont Auvergne, CNRS, SIGMA Clermont, Institut Pascal
 
-import copy
-from math import sqrt, tan, pi, cos, sin
-import matplotlib.pyplot as plt
+from math import pi, cos, sin
 from typing import Any
 from typing import List
+import matplotlib.pyplot as plt
 import numpy as np
-from nevergrad.common.typetools import ArrayLike
 from ... import instrumentation as inst
-from ...instrumentation.core import Variable
 from ...instrumentation.multivariables import Instrumentation
 
 
 class Agent():
-    """An agent has an input size, an output size, a number of layers, a width of its internal layers 
+    """An agent has an input size, an output size, a number of layers, a width of its internal layers
     (a.k.a number of neurons per hidden layer)."""
 
-    def __init__(self, input_size: int, output_size: int, layers: int=3, layer_width: int=14):
-        assert(layers >= 2)
+    def __init__(self, input_size: int, output_size: int, layers: int = 3, layer_width: int = 14) -> None:
+        assert layers >= 2
         self.input_size = input_size
         self.output_size = output_size
         self.layers: List[Any] = []
         self.layers += [np.zeros((input_size, layer_width))]
-        for i in range(layers-2):
+        for _ in range(layers - 2):
             self.layers += [np.zeros((layer_width, layer_width))]
         self.layers += [np.zeros((layer_width, output_size))]
         assert len(self.layers) == layers
 
-    def GetParamNumbers(self):
+    def GetParamNumbers(self) -> int:
         return sum([np.prod(l.shape) for l in self.layers])
 
-    def SetParams(self, ww):
+    def SetParams(self, ww: Any) -> None:
         w = [w for w in ww]
-        assert(len(w) == self.GetParamNumbers())
+        assert len(w) == self.GetParamNumbers()
         for l in self.layers:
             s = np.prod(l.shape)
-            l = np.reshape(np.array(w[:s]), l.shape)
+            l = np.reshape(np.array(w[:s]), l.shape)  # TODO @oteytaud new name?
```

Comment:
@teytaud this probably fails somehow since l is overridden by the for loop, can you check it?

Reply:
not sure, but I do a PR at least for readability

Reply:
actually I believe this was correct... in particular we get a decent comparison between algorithms in 
https://dl.fbaipublicfiles.com/nevergrad/allxps/list.html (powersystemsbig) which can not be due to random.

that said readability is a good reason

Reply:
I don't understand your answer.
My problem here is that this statement is pointless, since the value in l is lost when it is reset by the for loop. So either the line is not useful, or it is incorrect.